// tradingview-widget/esm2020/lib/tradingview-widget.component.mjs
import { Component, Input } from '@angular/core';
import { BarStyles, CONTAINER_ID, IntervalTypes, SCRIPT_ID, Themes } from './tradingview-widget.model';
import * as i0 from "@angular/core";
export class TradingviewWidgetComponent {
    constructor() {
        this._widgetConfig = {};
        this._defaultConfig = {
            symbol: 'NASDAQ:AAPL',
            allow_symbol_change: true,
            autosize: false,
            enable_publishing: false,
            height: 610,
            hideideas: true,
            hide_legend: false,
            hide_side_toolbar: true,
            hide_top_toolbar: false,
            interval: IntervalTypes.D,
            locale: 'en',
            save_image: true,
            show_popup_button: false,
            style: BarStyles.CANDLES,
            theme: Themes.LIGHT,
            timezone: 'Etc/UTC',
            toolbar_bg: '#F1F3F6',
            widgetType: 'widget',
            width: 980,
            withdateranges: false
        };
        this.style = {};
        this.containerId = CONTAINER_ID;
    }
    set widgetConfig(value) {
        this._widgetConfig = value;
        this.cleanWidget();
        this.initWidget();
    }
    get widgetConfig() {
        return this._widgetConfig || this._defaultConfig;
    }
    ngOnInit() {
        this.appendScript(this.initWidget.bind(this));
    }
    initWidget() {
        /* global TradingView */
        if (typeof TradingView === 'undefined' || !this.getContainer())
            return;
        const { widgetType, ...widgetConfig } = this.widgetConfig;
        const config = { ...widgetConfig, container_id: this.containerId };
        if (config.autosize) {
            delete config.width;
            delete config.height;
        }
        if (config.popup_width && typeof config.popup_width === 'number') {
            config.popup_width = config.popup_width.toString();
        }
        if (config.popup_height && typeof config.popup_height === 'number') {
            config.popup_height = config.popup_height.toString();
        }
        if (config.autosize) {
            this.style = {
                width: '100%',
                height: '100%'
            };
        }
        /* global TradingView */
        new TradingView[widgetType](config);
    }
    ;
    appendScript(onload) {
        if (!this.canUseDOM()) {
            onload();
            return;
        }
        if (this.scriptExists()) {
            /* global TradingView */
            if (typeof TradingView === 'undefined') {
                this.updateOnloadListener(onload);
                return;
            }
            onload();
            return;
        }
        const script = document.createElement('script');
        script.id = SCRIPT_ID;
        script.type = 'text/javascript';
        script.async = true;
        script.src = 'https://s3.tradingview.com/tv.js';
        script.onload = onload;
        document.getElementsByTagName('head')[0].appendChild(script);
    }
    ;
    canUseDOM() {
        return typeof window !== 'undefined' &&
            window.document &&
            window.document.createElement;
    }
    scriptExists() {
        return this.getScriptElement() !== null;
    }
    updateOnloadListener(onload) {
        const script = this.getScriptElement();
        const oldOnload = script.onload.bind(script);
        return script.onload = () => {
            oldOnload();
            onload();
        };
    }
    ;
    getScriptElement() {
        return document.getElementById(SCRIPT_ID);
    }
    cleanWidget() {
        if (!this.canUseDOM())
            return;
        const container = this.getContainer();
        if (container) {
            container.innerHTML = '';
        }
    }
    ;
    getContainer() {
        return document.getElementById(this.containerId);
    }
}
TradingviewWidgetComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.5", ngImport: i0, type: TradingviewWidgetComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
TradingviewWidgetComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.3.5", type: TradingviewWidgetComponent, selector: "tradingview-widget", inputs: { widgetConfig: "widgetConfig" }, ngImport: i0, template: `
    <section [id]="containerId"> </section>
  `, isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.5", ngImport: i0, type: TradingviewWidgetComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'tradingview-widget',
                    template: `
    <section [id]="containerId"> </section>
  `,
                    styles: []
                }]
        }], ctorParameters: function () { return []; }, propDecorators: { widgetConfig: [{
                type: Input,
                args: ['widgetConfig']
            }] } });
