// tradingview-widget/esm2020/lib/tradingview-widget.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { TradingviewWidgetComponent } from './tradingview-widget.component';
import * as i0 from "@angular/core";
export class TradingviewWidgetModule {
}
TradingviewWidgetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.5", ngImport: i0, type: TradingviewWidgetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
TradingviewWidgetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.3.5", ngImport: i0, type: TradingviewWidgetModule, declarations: [TradingviewWidgetComponent], imports: [CommonModule], exports: [TradingviewWidgetComponent] });
TradingviewWidgetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.3.5", ngImport: i0, type: TradingviewWidgetModule, providers: [], imports: [[
            CommonModule
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.5", ngImport: i0, type: TradingviewWidgetModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [TradingviewWidgetComponent],
                    imports: [
                        CommonModule
                    ],
                    exports: [TradingviewWidgetComponent],
                    providers: []
                }]
        }] });
